src/plugins/query_enhancements/public/search/ppl_search_interceptor.ts

```diff
@@ -95,6 +95,11 @@ export class PPLSearchInterceptor extends SearchInterceptor {
     const datasetService = queryString.getDatasetService();
     if (datasetService.getType(dataset.type)?.languageOverrides?.PPL?.hideDatePicker === false)
       return query;
+    if (
+      typeof query.query === 'string' &&
+      !query.query.toLowerCase().replace(/\s/g, '').startsWith('source=')
```

Comment:
i was thinking why you used `replace(/\s\g, '')` instead of just `trim()`, and then i realized that there can be spaces in between `source` and `=`. Can we extract this to a util function, like `queryStartsWithSource`? There's a lot going on especially in this already convoluted file that makes it difficult to read. (we can also unit test that util function, which supports the other comment by @ruanyl )